### ISSUE TITLE
Modulator: Envelope Generator

### ISSUE THREAD

**Issue by olofson** [olofson/audiality2#260]
Control register modulator that generates curves of various shapes, instead of the usual linear ramps built into (most) control registers.

Parameters:
- **target** - Target level (ramped control; durations optionally overridden by 'time')
- **time** - Ramp time override (ms; 0 means ramp timing follows 'target' ramping)
- **mode** - Output function for positive changes (step/linear/quadratic/cubic/inverse quadratic/...)
- **down** - Output function for negative changes (linked to 'mode'/step/linear/quadratic/cubic/inverse quadratic/...)

Depends on #51.

**Comment by olofson** [comment]
(Description updated to reflect the current state of the prototype in the wip-260 branch. It was originally in the wip-51 branch, as it was used as a pilot case for the control output API design.)

A remaining design issue is that plain x^N exponential curves beyond cubic aren't all that well suited for long decay/release envelopes, as they're too slow initially, and then get too close to the target level too early - so in practical terms, higher degrees just sound like faster envelopes.

The higher degree curves should behave more like natural decays. IIR filters multiplied with linear ramps, to actually hit the target...?

Also, for quick level changes with minimal transient artifacts, it would be nice to have one or two spline modes, both starting and ending with near-zero derivates.

**Comment by olofson** [comment]
TODO:
* Use mode/down 0 for LINK. (STEP is pretty useless, and can be performed via @/set anyway).
* Use -1 (current LINK code) for SPLINE.
* Investigate further what to do about the higher degree curves.

**Comment by olofson** [comment]
Also affected by #267, both directly when "forwarding" linear ramping, and through the internal ramper.
